internal/cmd/login.go

```diff
@@ -84,12 +68,15 @@ $ infra login`,
 				options.Server = args[0]
 			}
 
+			if user, ok := os.LookupEnv("INFRA_USER"); ok {
+				options.User = user
+			}
```

Comment:
The `cliopts.DefaultsFromEnv` I add in #3710 may be able to replace this.

I think this code may be wrong because the env var value overrides the cli flag, where as the usual order is that CLI flags override the env var.

By removing this, `cliopts.DefaultsFromEnv` will set the value correctly from env var, but only when the CLI flag is not set to a value.

Reply:
Great catch, thanks! Yes that's a bug – CLI flag should always take precedent if specified